app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,83 @@
+class DriversController < ApplicationController
+  def index
+    @drivers = Driver.all
+  end
+
+  def show
+    @driver = Driver.find_by(id: params[:id])
+
+    if @driver.nil?
+      redirect_to drivers_path
+      return
+    end
+
+    @trips = Trip.where(driver_id: @driver.id)
+    @passengers = Passenger.find_by(id: params[:id])
+
+  end
+
+  def new
+    @driver = Driver.new
+  end
+
+  def create
+    @driver = Driver.new(driver_params)
+    # default: a new driver is available
+    @driver[:available] = true
+    result = @driver.save
```

Comment:
You should try to be more specific with variable names than `result`.  

I would probably just put the `@driver.save` inline but if you want to store it in a variable I would name it something like `save_result`.